memote/support/helpers.py

```diff
@@ -693,28 +693,50 @@ def find_met_in_model(model, mnx_id, compartment_id=None):
         cobra.Metabolite(s) matching the mnx_id.
 
     """
+
+    def compare_annotation(annotation):
+        """Return IDs that are METANETX_SHORTLIST and annotation dictionary."""
+        query_values = set(utils.flatten(annotation.values()))
+        ref_values = set(utils.flatten(METANETX_SHORTLIST[mnx_id]))
+        return query_values & ref_values
+
+    # Make sure that the MNX ID we're looking up exists in the metabolite
+    # shortlist.
     if mnx_id not in METANETX_SHORTLIST.columns:
         raise RuntimeError("{} is not in the MetaNetX Shortlist! Make sure "
                            "you typed the ID correctly, if yes, update the "
                            "shortlist by updating and re-running the script "
                            "generate_mnx_shortlists.py.".format(mnx_id))
-
     candidates = []
-    regex = re.compile('^{}(_[a-zA-Z]+?)*?$'.format(mnx_id))
+    # The MNX ID used in the model may or may not be tagged with a compartment
+    # tag e.g. `MNXM23141_c` vs. `MNXM23141`, which is tested with the
+    # following regex.
+    # If the MNX ID itself cannot be found as an ID, we try all other
+    # identifiers that are provided by our shortlist of MetaNetX' mapping
+    # table.
+    regex = re.compile('^{}(_[a-zA-Z0-9]+?)*?$'.format(mnx_id))
```

Comment:
Since you're including the end of string mark `$` non-greedy matching should not be necessary. Also, we do expect at most one compartment suffix, right? My suggestion (untested)
`^{}(_[a-zA-Z0-9]+)?$` which should match exactly 0 or 1 underscore suffix.

Reply:
Done! Fingers crossed that all the tests still pass!